library/scripts/adi_ip.tcl

```diff
@@ -200,6 +201,32 @@ proc adi_ip_files {ip_name ip_files} {
   set_property "top" "$ip_name" $proj_fileset
 }
 
+# Ensure that the top file is the last one in the synthesis file group
+proc adi_ip_reorder_files {ip_name filegroup} {
+  # The last file in file group "Synthesis" should be an HDL file.
+  # During generation the IP Flow uses the last file to determine library and
+  # other information when generating the top wrapper file.
+  set ip_files_for_syn [ipx::get_files -of $filegroup]
+  set top_file ""
+  # Find the top level file
+  foreach cur_file $ip_files_for_syn {
+    set filename [get_property NAME $cur_file]
+    if { [file rootname [file tail $filename]] == $ip_name &&
+         ([file extension [file tail $filename]] == ".v"  ||
+          [file extension [file tail $filename]] == ".sv" ||
+          [file extension [file tail $filename]] == ".vhd") } {
+      set top_file $filename
+      break
+    }
+  }
+  if { $top_file != "" } {
+    # Place the topfile after the last file from the filegroup
+    set lastfile [get_property NAME [lindex $ip_files_for_syn end]]
+    ipx::reorder_files -after $lastfile $top_file $filegroup
```

Comment:
reorder_files has a -back option. Does that work instead of using -after?

Reply:
Yes, that works, thanks for the suggestion.